app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt

```diff
@@ -1,207 +1,251 @@
+                            onConfirmClick = {
+                                if (it > 0) {
+                                    pickedSleepTimer(it)
+                                }
+                            },
+                        )
+                    }
+                }
 
-            flashlightBtn.setOnClickListener {
-                mCameraImpl!!.toggleFlashlight()
-            }
+                val sleepTimerDialogState = rememberAlertDialogState().apply {
+                    DialogMember {
+                        SleepTimerRadioDialog(
+                            alertDialogState = this,
+                            customAlertDialogState = sleepTimerCustomDialogState
+                        )
+                    }
+                }
 
-            sosBtn.setOnClickListener {
-                toggleStroboscope(true)
-            }
+                val sleepTimerPermissionDialogState = rememberAlertDialogState().apply {
+                    DialogMember {
+                        PermissionRequiredAlertDialog(
+                            alertDialogState = this,
+                            text = stringResource(id = R.string.allow_alarm_sleep_timer),
+                            positiveActionCallback = {
+                                openRequestExactAlarmSettings(baseConfig.appId)
+                            },
+                            negativeActionCallback = {
+                                sleepTimerDialogState.show()
+                            }
+                        )
+                    }
+                }
 
-            stroboscopeBtn.setOnClickListener {
-                toggleStroboscope(false)
+                MainScreen(
+                    flashlightButton = {
+                        val flashlightActive by viewModel.flashlightOn.collectAsStateWithLifecycle()
+
+                        FlashlightButton(
+                            onFlashlightPress = { viewModel.toggleFlashlight() },
+                            flashlightActive = flashlightActive,
+                        )
+                    },
+                    brightDisplayButton = {
+                        val showBrightDisplayButton by preferences.brightDisplayFlow.collectAsStateWithLifecycle(
+                            config.brightDisplay,
+                            minActiveState = Lifecycle.State.CREATED
+                        )
+                        if (showBrightDisplayButton) {
+                            BrightDisplayButton(
+                                onBrightDisplayPress = {
+                                    startActivity(Intent(applicationContext, BrightDisplayActivity::class.java))
+                                }
+                            )
+                        }
+                    },
+                    sosButton = {
+                        val showSosButton by preferences.sosFlow.collectAsStateWithLifecycle(config.sos, minActiveState = Lifecycle.State.CREATED)
```

Comment:
Is there a particular reason why this minActiveState needs to be CREATED?

Reply:
and also on line 166?

Reply:
I am guessing I messed something up with recomposition, because when I remove that, going to settings and then changing the displayed state of either of these buttons is not properly caught and they keep their old state.

Reply:
try moving that line outside of the `sosButton` lambda composable